db/flush_scheduler.h

```diff
@@ -45,4 +50,58 @@ class FlushScheduler {
 #endif  // NDEBUG
 };
 
+// TODO (yanqin) move to a different file.
+class InternalFlushManager : public FlushManager {
```

Comment:
Inheritance not necessary.

Reply:
We would like `NewFlushManager` to return a pointer to `FlushManager` while the underlying object is `DefaultFlushManager` or some other class that share some common internal interface `InternalFlushManager`.

Reply:
I have removed this inheritance.